leasing/models/invoice.py

```diff
@@ -18,3 +18,14 @@ class Invoice(TimestampedModelMixin):
                                         blank=True, on_delete=models.PROTECT)
     state = EnumField(InvoiceState, verbose_name=_("State"), max_length=255,
                       default=InvoiceState.PENDING)
+
+    def create_reference_number(self):
+        if not self.id:
+            return None
+
+        reference_number = '91112{}880'.format(self.id)
+
+        self.reference_number = reference_number + str(
+            (10 - (sum((7, 3, 1)[i % 3] * x for i, x in enumerate(map(int, str(reference_number)[::-1]))) % 10)) % 10)
```

Comment:
OK, I also optimized away the extra `% 10` and replaced 10 with zero (since 10 == 0 mod 10).